lib/Runtime/Language/JavascriptConversion.cpp

```diff
@@ -450,20 +450,21 @@ namespace Js
     //4. If SameValue(ToString(n), argument) is false, then return undefined.
     //5. Return n.
     //----------------------------------------------------------------------------
-    BOOL JavascriptConversion::CanonicalNumericIndexString(Var aValue, double *indexValue, ScriptContext * scriptContext)
+    BOOL JavascriptConversion::CanonicalNumericIndexString(JavascriptString *aValue, double *indexValue, ScriptContext * scriptContext)
     {
-        AssertMsg(JavascriptString::Is(aValue), "CanonicalNumericIndexString expects only string");
-        if (JavascriptString::IsNegZero(JavascriptString::FromVar(aValue)))
+        if (JavascriptString::IsNegZero(aValue))
         {
             *indexValue = -0;
             return TRUE;
         }
-        Var indexNumberValue = JavascriptOperators::ToNumber(aValue, scriptContext);
-        if (JavascriptString::Equals(JavascriptConversion::ToString(indexNumberValue, scriptContext), aValue))
+
+        double indexDoubleValue = aValue->ToDouble();
+        if (JavascriptString::Equals(JavascriptNumber::ToStringRadix10(indexDoubleValue, scriptContext), aValue))
```

Comment:
Seems like this would be faster, but just want to check whether that's the motivation (or if there's some other motivation) behind changing from from converting String(Var)->Number(Var)->String(Var), to instead String(Var)->double(native)->String(Var). Any potential correctness concerns?